src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.cs

```diff
@@ -454,17 +454,27 @@ public static void FileSystemWatcher_CreateManyConcurrentInstances()
     {
         int maxUserInstances = int.Parse(File.ReadAllText("/proc/sys/fs/inotify/max_user_instances"));
         var watchers = new List<FileSystemWatcher>();
+        var directories = new List<string>();
 
         using (var dir = Utility.CreateTestDirectory())
         {
+            // We shouldn't catch I/O exception here since this is test setup
```

Comment:
True...what would you propose here instead? Catch the exception and retry creating the directories?